File: android/src/main/kotlin/com/shpak/subscription_checker/model/CheckResult.kt
package com.shpak.subscription_checker.model

data class CheckResult(
    val subscriptionStatus: SubscriptionStatus,
    val purchaseTimeMillis: Long? = null
) {
    val asMap: Map<String, Any?>
        get() = mapOf(
            "subscriptionStatus" to subscriptionStatus.status,
            "purchaseTimeMillis" to purchaseTimeMillis
        )
}

File: lib/model/check_result.dart
import 'package:subscription_checker/model/subscription_status.dart';

class CheckResult {
  CheckResult({required this.subscriptionStatus, this.purchaseTimeMillis});

  final SubscriptionStatus subscriptionStatus;
  final int? purchaseTimeMillis;

  @override
  String toString() {
    return 'CheckResult{subscriptionStatus: $subscriptionStatus, purchaseTimeMillis: $purchaseTimeMillis}';
  }
}
